### DISCUSSION TITLE
Middleware: How to rewrite api with authorization header to another api end point?

### DISCUSSION THREAD

**Discussion by manoj-mukherjee-maersk** [vercel/next.js#33492] [Help] [answered]
We are using Nextjs 12. Able to rewrite API endpoint another endpoint but need to add authorization header from middleware for security purpose.

Is it possible to add req header before rewrite?

**Comment by leerob** [comment, marked as answer]
https://github.com/vercel/examples/blob/main/edge-functions/add-header/pages/_middleware.ts

**Reply by manoj-mukherjee-maersk** [reply]
Looking for adding request header appending. not response header.
